crates/dojo-lang/src/interface.rs

```diff
@@ -0,0 +1,197 @@
+use std::collections::HashMap;
+
+use cairo_lang_defs::patcher::{PatchBuilder, RewriteNode};
+use cairo_lang_defs::plugin::{
+    DynGeneratedFileAuxData, PluginDiagnostic, PluginGeneratedFile, PluginResult,
+};
+use cairo_lang_diagnostics::Severity;
+use cairo_lang_syntax::node::ids::SyntaxStablePtrId;
+use cairo_lang_syntax::node::{ast, Terminal, TypedSyntaxNode};
+use cairo_lang_syntax::node::ast::{MaybeTraitBody, OptionReturnTypeClause, ParamList};
+use cairo_lang_syntax::node::db::SyntaxGroup;
+use cairo_lang_utils::unordered_hash_map::UnorderedHashMap;
+use dojo_types::system::Dependency;
+
+use crate::plugin::{DojoAuxData, SystemAuxData};
+
+pub struct DojoInterface {
+    diagnostics: Vec<PluginDiagnostic>,
+    dependencies: HashMap<smol_str::SmolStr, Dependency>,
+}
+
+impl DojoInterface {
+    pub fn from_trait(db: &dyn SyntaxGroup, trait_ast: ast::ItemTrait) -> PluginResult {
+        let name = trait_ast.name(db).text(db);
+        let mut system = DojoInterface { diagnostics: vec![], dependencies: HashMap::new() };
+        let mut builder = PatchBuilder::new(db);
+
+        if let MaybeTraitBody::Some(body) = trait_ast.body(db) {
+            let body_nodes: Vec<_> = body
+                .items(db)
+                .elements(db)
+                .iter()
+                .flat_map(|el| {
+                      if let ast::TraitItem::Function(fn_ast) = el {
+                        return system.rewrite_function(db, fn_ast.clone());
+                    }
+                    
+                    //TODO: if the trait body contains other things than functions, 
```

Comment:
Hm is anything other than functions valid in a trait? If so, we should at least output a diagnostic here to say it is not supported yet

Reply:
Ok, I did some tests and if I add a const or an enum in a `[starknet::interface]` trait I got an error. So, let's say we cannot have anything other than functions in a `[starknet::interface]`. 
Yes @tarrencev, I will output a diagnostic in this case 👍